### PR TITLE
feat: Use vuex for the global store

### PR DESCRIPTION
**Pull request by kinnarr**
This needs still some usage testing

### CODE REVIEW

File: src/store/metrics.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+        drawMax: metric.drawMax
+      }
+    },
+    getAllMetricKeys: (state) => () => {
+      return Object.keys(state.metrics)
+    },
+    getAllMetrics: (state) => () => {
+      return Object.values(state.metrics)
+    }
+  },
```
Comment (tilsche) [inline src/store/metrics.js:29]: Why is `metrics.metrics` accessed without a getter so often? That seems inconsistent.

File: src/store/metrics.js [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+  mutations: {
+    resetAllMetrics (state) {
+      console.log('Reset allMetrics')
+      Object.keys(state.metrics).forEach(function (key) { if (!key.startsWith('_')) { Vue.delete(state.metrics, key) } })
+    },
+    removeMetricEntry (state, { metricKey }) {
```
Comment (tilsche) [inline src/store/metrics.js:35]: just... `deleteMetric`?

File: src/store/metrics.js [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+          Vue.set(state.metrics[metricKey], 'drawMax', drawMax)
+        }
+      }
+    },
+
+    setMetricPopup (state, { metricKey, popupState }) {
+      if (state.metrics[metricKey]) {
+        Vue.set(state.metrics[metricKey], 'popup', popupState)
+      }
+    }
```
Comment (tilsche) [inline src/store/metrics.js:95]: else should probably be an error?

**Review by tilsche — COMMENTED** [review]
Partial review for today with a focus on the store itself.

File: src/store/metrics.js [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+      if (getters.getMetricKey(newKey) !== undefined) {
+        console.log('Metric already exists!')
+        return
+      }
+      const oldMetric = state.metrics[oldKey]
+      commit('setMetric', { metricKey: newKey, metric: oldMetric })
+      const newMetric = state.metrics[newKey]
+      dispatch('updateColor', { metricKey: name, color: newMetric.color })
+      dispatch('updateMarker', { metricKey: name, color: newMetric.marker })
+      dispatch('updateDescription', { metricKey: name, description: newMetric.description })
```
[the remark this answers is not in the document]
Comment (kinnarr) [reply src/store/metrics.js]: `name` was a copy-paste error. 
For the use of `metrics/createMetric` we need to allow more attributes to get an exact copy. At the moment `metrics/createMetric` only supports `name`, `description` and `traces`

File: src/store/metrics.js (hunk 2 above)
Comment (kinnarr) [reply src/store/metrics.js:35]: There is already an action `deleteMetric` which calls `removeMetricEntry` and `checkMetricDrawState`

**Review by tilsche — COMMENTED** [review]
another batch of comments

File: src/store/metrics.js [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+      const newMetric = state.metrics[newKey]
+      dispatch('updateColor', { metricKey: newKey, color: newMetric.color })
+      dispatch('updateMarker', { metricKey: newKey, color: newMetric.marker })
+      dispatch('updateDescription', { metricKey: newKey, description: undefined })
+    },
+    updateDescription ({ commit, state }, { metricKey, description }) {
+      const metric = state.metrics[metricKey]
+      if (description === undefined) {
+        commit('setMetric', { metricKey, metric: { description: '' } })
+        window.MetricQWebView.instances[0].handler.metricQHistory.metadata(metric.name).then((metadataObj) => {
```
[the remark this answers is not in the document]
Comment (tilsche) [reply src/store/metrics.js:136]: not yet

File: src/ui/configuration-popup.vue [hunk 6]
```diff
@@ -148,20 +136,20 @@ export default {
...
     uiZoomSpeed: {
       cache: false,
       get: function () {
-        return this.config.zoomSpeed
+        return this.configuration.zoomSpeed
       },
       set: function (newValue) {
-        Store.state.configuration.zoomSpeed = newValue
+        this.$store.commit('setZoomSpeed', newValue)
         this.$emit('update:uiZoomSpeed', newValue)
```
[the remark this answers is not in the document]
Comment (tilsche) [reply src/ui/configuration-popup.vue:153]: noone knows -> remove it

File: src/ui/configuration-popup.vue [hunk 7]
```diff
@@ -148,20 +136,20 @@ export default {
     uiResolution: {
       cache: false,
       get: function () {
-        return 30 - this.config.resolution
+        return 30 - this.configuration.resolution
       },
       set: function (newValue) {
-        Store.state.configuration.resolution = 30 - newValue
+        this.$store.commit('setResolution', 30 - newValue)
         this.$emit('update:uiResolution', newValue)
```
[the remark this answers is not in the document]
Comment (tilsche) [reply src/ui/configuration-popup.vue:143]: noone knows -> remove it

File: src/store/metrics.js [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+
+      const metric = state.metrics[metricKey]
+      dispatch('updateColor', { metricKey, color: metric.color })
+      dispatch('updateMarker', { metricKey, marker: metric.marker })
+    },
+    setMetricError ({ commit, state }, { metricKey }) {
+      const metric = state.metrics[metricKey]
+      commit('setMetric', { metricKey: metricKey, metric: { name: metric.name, errorprone: true } })
+    },
+    updateDrawMinMaxGlobal ({ state, commit }, newState) {
```
[the remark this answers is not in the document]
Comment (kinnarr) [reply src/store/metrics.js:194]: How about `updateDrawStateGlobally`?

**Pull request merged**